server/aws-lsp-codewhisperer/src/language-server/pathUtils.ts

```diff
@@ -0,0 +1,51 @@
+import os = require('os')
+
+/**
+ * Returns `true` if path `p` is a descendant of directory `d` (or if they are
```

Comment:
ideally as a followup it would be nice to have some unit tests for these functions